src/components/ProductCard/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { css } from '@emotion/css';
+import RemoveButton from '../Button/RemoveButton';
+import AddButton from '../Button/AddButton';
+import { Product } from '../../types/product.type';
+import { useShoppingCartContext } from '../../contexts/useShoppingCartContext';
+import { useAddShoppingCart } from '../../hooks/useAddShoppingCart';
+import { useDeleteShoppingCart } from '../../hooks/useDeleteShoppingCart';
+
+interface ProductCardProps {
+  product: Product;
+  isInCart: boolean;
+}
+
+const ProductCard = ({ product, isInCart }: ProductCardProps) => {
+  const { id, name, price, imageUrl } = product;
+  const shoppingCart = useShoppingCartContext();
+  const cartItemId = shoppingCart.items.find(
+    (item) => item.product.id === product.id
+  )?.id;
+  const addShoppingCart = useAddShoppingCart(product.id);
+  const deleteShoppingCart = useDeleteShoppingCart(cartItemId);
```

Comment:
useDeleteShoppingCart와 useAddShoppingCart 내부에서 useShoppingCartContext를 사용하도록 되어있어요.

그런데 이 코드들이 쓰이는 곳에서 또 useShoppingCartContext를 사용하고 있습니다.

저는 이런 모습을 "추상화 수준이 다르다" 라고 표현하는데요,

1. 전부다 밖에서 쓰이는 모습으로 만들어지거나
2. 전부다 안에서 쓰이는 모습으로 만들어지거나

이렇게 일관성 있는 모습이 필요하다고 생각해요

가령, 이렇게 하나의 훅을 더 만들어주는거죠.

```suggestion
  const cartItem = useShoppingCartItem(product.id);
  const addShoppingCart = useAddShoppingCart(product.id);
  const deleteShoppingCart = useDeleteShoppingCart(cartItem?.Id);
```

더 나아가서, 아예 이렇게 세 개의 훅을 호출하는 훅을 하나 더 만들 수도 있겠죠?

```tsx
  const shoppingCart = useShoppingCart(product.id);
  shoppingCart.item
  shoppingCart.update()
  shoppingCart.delete()
```
요로코롬 쓰일 수 있도록!